### DISCUSSION TITLE
Watermark PDF Multi-page

### DISCUSSION THREAD

**Discussion by neildonnell** [ImageMagick/ImageMagick#6898] [Help]
### ImageMagick version

7

### Operating system, version and so on

Windows

### Description

I have the following command that has been serving me well for watermarking images and single page PDF's:

`magick.exe a.pdf ^( -size 200x -background none -fill "RGBA(200,255,255,0.2)" -gravity center "label:%s_watermark%" -trim -rotate -30 -bordercolor none -border 10 -write mpr:wm +delete +clone -fill mpr:wm  -draw  "color 0,0 reset" ^) -compose over -composite output.pdf`

But today I found it didn't work on multi-page pdf's.  It just returns one page without a watermark.  I have tried modifying the command but now this returns a lower quality and non-watermarked pdf.   Can someone advise on where I might be going wrong?  I am assuming it's in the compose state that something is empty but I can't work out why/how.

`magick.exe input.pdf null: ^( -size 200x -background none -fill "RGBA(200,255,255,0.2)" -gravity center "label:%s_watermark%" -trim -rotate -30 -bordercolor none -border 10 -write mpr:wm +delete +clone -fill mpr:wm  -draw  "color 0,0 reset" ^) -compose multiply -layers composite output.pdf`

**Comment by snibgo** [comment]
If `-compose Over` does what you want, why did you change that to `-compose Multiply`?

Within the parentheses, you have `+clone`. This copies the final image from the list before the parenthesis. In your first command, that image is the last page of the PDF. But in your revised command, that image is `null:`. I suggest using `-clone -2` or `-clone 0` instead.

**Comment by nicolnt** [comment]
I want to repeat a text watermark on all the pages of a PDF.
I tried to adapt the code of @neildonnell as well as this one here https://stackoverflow.com/questions/72586533/how-to-create-a-watermark-aligned-at-an-angle#72586824

I have this code below, which exports all the pages of the PDF.
However, the watermark is repeated only once and on the first page (top-left corner), see screenshot below.
![image](https://github.com/user-attachments/assets/94cb0456-2e1c-47f2-80ec-a5468ba0270d)

Example document: 
[document.pdf](https://github.com/user-attachments/files/18149202/document.pdf)


Can someone help me get what I need: repeated watermark generated by text on all the pages of a PDF.

```bash
convert \
document.pdf \
null: \( -background None -fill "#fff7" -pointsize 20 label:"My watermark" \
-rotate -40 +repage +write mpr:wm +clone -fill mpr:wm -draw "color 0,0 reset" \) \
-compose over -layers composite output.pdf
```

**Comment by neildonnell** [comment]
This is what I ended up doing.   I manually do a number of lines and repeat the watermark.  This has given me the best results.


`magick.exe" -density 300 "1.0.pdf" -background none -fill "rgba(200, 200, 200, 0.3)" -pointsize 32 -gravity center -draw "translate 0,0 rotate -45 text 0,0 ""WATERMARK WATERMARK WATERMARK WATERMARK WATERMARK WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate -250,-250 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate -500,-500 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate 1000,1000 rotate -45 text 2,2 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw  "translate 250,250 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate 500,500 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate 750,750 rotate -45 text 2,2 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw  "translate -1250,-1250 rotate -45 text 2,2 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate 1250,1250 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate -750,-750 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -fill "rgba(200, 200, 200, 0.3)" -gravity center -draw "translate -1000,-1000 rotate -45 text 0,0 ""WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK  WATERMARK""" -density 300 -quality 100 "1.0_watermarked.pdf""`

**Comment by fmw42** [comment]
Try this

[barn.pdf](https://github.com/user-attachments/files/18154360/barn.pdf)

Unix syntax:

```
magick \
-background None -fill "#fff7" \
-pointsize 20 label:"Watermark" \
-rotate -40 +repage \
-write mpr:TILE +delete \
\( barn.pdf[0] -fill mpr:TILE \
-draw 'color 0,0 reset' -write mpr:TILES +delete \) \
barn.pdf -coalesce NULL: \
mpr:TILES \
-compose over -layers composite \
barn_watermarked.pdf

```

Windows syntax

```
magick ^
-background None -fill "#fff7" ^
-pointsize 20 label:"Watermark" ^
-rotate -40 +repage ^
-write mpr:TILE +delete ^
( barn.pdf[0] -fill mpr:TILE ^
-draw 'color 0,0 reset' -write mpr:TILES +delete ) ^
barn.pdf -coalesce NULL: ^
mpr:TILES ^
-compose over -layers composite ^
barn_watermarked.pdf

```

[barn_watermarked.pdf](https://github.com/user-attachments/files/18154376/barn_watermarked.pdf)

**Reply by nicolnt** [reply]
Thanks,
I tried this code on my document!
You are really close to what I am looking for, but I don't understand why the watermark is not printed on the whole page though (see screenshot below)

## EDIT

That's perfect actually, many thanks!!

I did not catch that the barn PDF was mentioned in two places, so it was using `barn.pdf[0]` instead of `document.pdf[0]`

![image](https://github.com/user-attachments/assets/a2765446-cd80-41f0-ac64-14fa67dbebbf)
